crates/module-system/module-implementations/sov-evm/src/call.rs

```diff
@@ -123,6 +124,36 @@ where
         self.pending_transactions
             .push(&pending_transaction, state)?;
 
+        // Fetch `head` and `pending_len` before the `native` code block.
+        // This ensures consistent gas charges between native and non-native execution.
+        #[allow(unused_variables)]
+        let head = self
+            .head
+            .get(state)?
+            .expect("Impossible happened: Head must be set.");
+
+        #[allow(unused_variables)]
+        let pending_len = self.pending_transactions.len(state)?;
+
+        #[cfg(feature = "native")]
+        {
+            let first_tx_index = head.transactions.end;
+            let tx_index = first_tx_index + pending_len - 1;
+
+            self.transactions
+                .set(&tx_index, &pending_transaction.transaction, state)
+                .unwrap_infallible();
```

Comment:
This function returns anyhow::Result<()> so we can use ? here which is more concise

Reply:
Will address in folloup